tests/Unit/SccmIniFIle.tests.ps1

```diff
@@ -0,0 +1,322 @@
+                            $combined = $test + $optionalParamatersAll + $primaryInstallOptionalSettings
+                            Set-TargetResource @combined
+                            Get-Item -LiteralPath "$IniFilePath/$IniFileName" | Should -Not -Be $null
+                        }
+                    }
+                }
+
+                Context "When Set-TargetResource for $($test.IniFileName) has incorrect parameters" {
+                    if ($test.iniFileName -eq 'InstallCas.ini')
+                    {
+                        $wrongParameters = $test.clone()
+                        $wrongParameters.Add('ManagementPoint','PRI.contoso.com')
+                        $message = 'The parameters ManagementPoint, ManagementPointProtocol, DistributionPoint,',
+                            'DistributionPointProtocol, RoleCommunicationProtocol, ClientsUsePKICertificate,',
+                        'CCARSiteServer, CASRetryInterval, WaitForCASTimeout are used only with InstallPrimarySite.'
+                        $throwreason = 'Wrong parameter for CAS'
+
+                        It "Should throw because $throwReason" {
+                            { Set-TargetResource @wrongParameters } | Should -Throw -ExpectedMessage $message
+                        }
+                    }
+                    if ($test.iniFileName -eq 'InstallPrimary.ini')
+                    {
+                        $wrongParameters = $test.Clone()
+                        $wrongParameters.CloudConnector = $true
+                        $wrongParameters.Add('CloudConnectorServer','CAS.contoso.com')
+                        $message = 'If CloudConnector is True you must provide CloudConnectorServer and UseProxy.'
+                        $throwReason = 'Missing Cloud Connector parameter'
+
+                        It "Should throw because $throwReason" {
+                            { Set-TargetResource @wrongParameters } | Should -Throw -ExpectedMessage $message
+                        }
+                    }
+
+                    It '"Should throw because missing proxy parameters' {
+                        $proxyTest = $test.Clone()
+                        $proxyTest.Add('UseProxy', $true)
+                        { Set-TargetResource @proxyTest } | Should -Throw -ExpectedMessage 'If Proxy is True, you must provide ProxyName and ProxyPort.'
+                    }
+                }
+            }
+        }
+
+        Describe "$moduleResourceName\Test-TargetResource" {
+            foreach ($test in $tests)
+            {
+                # Order of the Context blocks matter since some attributes are being changed in different Context.
+                Context "When ini file does not exist for $($test.IniFileName)" {
+                    It 'Should return false' {
+                        $result = Test-TargetResource @test
+                        $result | Should -Be $false
```

Comment:
don't really need $result could just be
Test-TargetResource | Should -be $false

Reply:
Good call. Changing in my next sync.